### ISSUE TITLE
Confusing error when destructuring object

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#420]
```js
    let queue = ["first", "second", "third"];
    let {first: 0} = queue;
```
expected:
> hello.js:7:17: error: expected variable name
observed:
> hello.js:7:17: error: invalid function parameter [E151]
